### PR TITLE
add 21. Merge Two Sorted Lists

### PR DESCRIPTION
**Pull request by katsurada1**
Please check my code.

### CODE REVIEW

File: 21_Merge_Two_Sorted_Lists.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+        if l2 is None:
+            return l1
+
+        if l1.val < l2.val:
+            temp = head = ListNode(l1.val)
+            l1 = l1.next
+        else:
+            temp = head = ListNode(l2.val)
+            l2 = l2.next
```
Comment (kobakobashu) [inline 21_Merge_Two_Sorted_Lists.py:32]: temp = head = ListNode(0)
l49 -> return head.next

File: 21_Merge_Two_Sorted_Lists.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# n = len(l1)
+# m = len(l2)
+# time: O(n + m)
+# space: O(1)
+
+# Definition for singly-linked list.
+# class ListNode(object):
+#     def __init__(self, val=0, next=None):
+#         self.val = val
+#         self.next = next
```
Comment (kobakobashu) [inline 21_Merge_Two_Sorted_Lists.py:10]: Please open two lines.

File: 21_Merge_Two_Sorted_Lists.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        if l2 is None:
+            return l1
+
+        if l1.val < l2.val:
+            temp = head = ListNode(l1.val)
+            l1 = l1.next
+        else:
+            temp = head = ListNode(l2.val)
+            l2 = l2.next
+        while l1 is not None and l2 is not None:
```
Comment (kobakobashu) [inline 21_Merge_Two_Sorted_Lists.py:33]: while l1 and l2

**Review by kobakobashu — CHANGES_REQUESTED** [review]
👀

File: 21_Merge_Two_Sorted_Lists.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            if l1.val < l2.val:
+                temp.next = ListNode(l1.val)
+                l1 = l1.next
+            else:
+                temp.next = ListNode(l2.val)
+                l2 = l2.next
+            temp = temp.next
+        while l1 is not None:
```
Comment (kobakobashu) [inline 21_Merge_Two_Sorted_Lists.py:41]: while l1

**Comment by katsurada1** [comment]
I am sorry I forgot to correct the code after I was taught on Sunday.
I modified the code.
Please check my code.

**Comment by kobakobashu** [comment]
LGTM. Please delete the original file and merge only the new one

**Review by kobakobashu — APPROVED** [review]

**Pull request merged**